--- vdrold/osdbase.c	2004-06-13 13:52:47.000000000 +0200
+++ vdr-1.2.6/osdbase.c	2004-06-13 13:49:23.000000000 +0200
@@ -116,7 +116,7 @@
   fontType = fontOsd;
   font = NULL;
   if (width > 0 && height > 0) {
-     bitmap = MALLOC(char, width * height);
+     bitmap = MALLOC(unsigned char, width * height);
      if (bitmap) {
         Clean();
         memset(bitmap, 0x00, width * height);
@@ -188,7 +188,7 @@
   dirtyY2 = -1;
 }
 
-void cBitmap::SetIndex(int x, int y, char Index)
+void cBitmap::SetIndex(int x, int y, unsigned char Index)
 {
   if (bitmap) {
      if (0 <= x && x < width && 0 <= y && y < height) {
@@ -269,7 +269,7 @@
      Fill(0, 0, width - 1, height - 1, clrBackground);
 }
 
-const char *cBitmap::Data(int x, int y)
+const unsigned char *cBitmap::Data(int x, int y)
 {
   return &bitmap[y * width + x];
 }
@@ -329,7 +329,7 @@
   cBitmap::Text(x, y, s, ColorFg, ColorBg);
 }
 
-const char *cWindow::Data(int x, int y)
+const unsigned char *cWindow::Data(int x, int y)
 {
   return cBitmap::Data(x, y);
 }
--- vdrold/osdbase.h	2004-06-13 13:52:47.000000000 +0200
+++ vdr-1.2.6/osdbase.h	2004-06-13 13:48:49.000000000 +0200
@@ -13,7 +13,7 @@
 #include <stdio.h>
 #include "font.h"
 
-#define MAXNUMCOLORS 16
+#define MAXNUMCOLORS 256
 
 enum eDvbColor {
 #ifdef DEBUG_OSD
@@ -75,7 +75,7 @@
 private:
   cFont *font;
   eDvbFont fontType;
-  char *bitmap;
+  unsigned char *bitmap;
   bool clearWithBackground;
 protected:
   int width, height;
@@ -86,7 +86,7 @@
   bool ClearWithBackground(void) { return clearWithBackground; }
   eDvbFont SetFont(eDvbFont Font);
   bool Dirty(int &x1, int &y1, int &x2, int &y2);
-  void SetIndex(int x, int y, char Index);
+  void SetIndex(int x, int y, unsigned char Index);
   void SetPixel(int x, int y, eDvbColor Color);
   void SetBitmap(int x, int y, const cBitmap &Bitmap);
   int Width(void) { return width; }
@@ -97,7 +97,7 @@
   void Fill(int x1, int y1, int x2, int y2, eDvbColor Color);
   void Clean(void);
   void Clear(void);
-  const char *Data(int x, int y);
+  const unsigned char *Data(int x, int y);
   };
 
 #define MAXNUMWINDOWS 7 // OSD windows are counted 1...7
@@ -122,7 +122,7 @@
   void Fill(int x1, int y1, int x2, int y2, eDvbColor Color);
   void SetBitmap(int x, int y, const cBitmap &Bitmap);
   void Text(int x, int y, const char *s, eDvbColor ColorFg = clrWhite, eDvbColor ColorBg = clrBackground);
-  const char *Data(int x, int y);
+  const unsigned char *Data(int x, int y);
   };
 
 typedef int tWindowHandle;
